### PR TITLE
[FEATURE] Generate the AppCacheBuster index file for apps

### PR DESCRIPTION
**Pull request by petermuessig**
**Thank you for your contribution!** 🙌

To get it merged faster, kindly review the checklist below:

## Pull Request Checklist
- [x] Reviewed the [Contributing Guidelines](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#-contributing-code)
    + Especially the [How to Contribute](https://github.com/SAP/ui5-tooling/blob/master/CONTRIBUTING.md#how-to-contribute) section 
- [x] [No merge commits](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#no-merge-commits)
- [x] [Correct commit message style](https://github.com/SAP/ui5-tooling/blob/master/docs/Guidelines.md#commit-message-style)

### CODE REVIEW

File: lib/tasks/generateCachebusterInfo.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	return workspace.byGlob(`/resources/${options.namespace}/**/*`)
+		.then(async (resources) => {
+			const cachebusterInfo = {};
+			const regex = new RegExp(`^/resources/${options.namespace}/`);
+			resources.forEach((resource) => {
+				const normalizedPath = resource.getPath().replace(regex, "");
+				cachebusterInfo[normalizedPath] = resource.getStatInfo().mtime.getTime();
+			});
+			const cachebusterInfoResource = resourceFactory.createResource({
+				path: "/sap-ui-cachebuster-info.json",
```
[the remark this answers is not in the document]
Comment (petermuessig) [reply lib/tasks/generateCachebusterInfo.js]: Fixed!

File: lib/types/application/ApplicationBuilder.js [hunk 2]
```diff
@@ -103,6 +104,17 @@ class ApplicationBuilder extends AbstractBuilder {
 			});
 		});
 
+		this.addTask("generateCachebusterInfo", () => {
```
[the remark this answers is not in the document]
Comment (RandomByte) [reply lib/types/application/ApplicationBuilder.js]: You also need to adapt the tests for this change 👍   
See the failing travis builds

File: lib/tasks/generateCachebusterInfo.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RandomByte) [reply lib/tasks/generateCachebusterInfo.js]: In that case it should actually be written to `/resources/${project.metadata.namespace}/sap-ui-cachebuster-info.json`. The applications resources will be re-mapped to the root of the `dist` directory. 

You can compare this to the applications `index.html`, which is expected in the `dist` root directory as well.

[the remark this answers is not in the document]
Comment (petermuessig) [reply lib/tasks/generateCachebusterInfo.js]: Oh, yes - I missed that part...

**Comment by petermuessig** [comment]
generateCachebusterInfo is now an optional task which needs to be enabled via includeTasks in the commandline

**Comment by RandomByte** [comment]
Please squash the two commits that have an identical message.

**Pull request merged**